crowdin.yml

```diff
@@ -0,0 +1,27 @@
+base_path: "."
+base_url: "https://neoforged.api.crowdin.com"
+
+# For interacting with Crowdin locally (via CLI), set these environment variables to their proper values
+project_id_env: "NEOFORGE_CROWDIN_PROJECT_ID"
+api_token_env: "NEOFORGE_CROWDIN_API_TOKEN"
+
+# Preserve directory structure (required by the use of the 'dest' field below)
+preserve_hierarchy: true
+
+# Prevent Crowdin-created commits from automatically triggering builds
+commit_message: "[ci skip]"
+# Automatically label PRs made by Crowdin
+pull_request_labels: [ "l10n" ]
+
+files:
+    -   source: "/src/main/resources/assets/neoforge/lang/en_us.json"
+        # Unfortunately, Crowdin doesn't have a placeholder which matches Minecraft's expected naming scheme: underscores
+        # with lowercase letters (since resource location are required to be fully lowercase).
+        #
+        # To make up for this unfortunate defect, **in the Crowdin interface**, we specify custom language mappings for
+        # the "%locale_with_underscore%" placeholder, converting the uppercase to lowercase in the original value.
+        #
+        # If you are adding a new language to the Crowdin project,
+        # ** MAKE SURE YOU ADD THE APPROPRIATE CUSTOM LANGUAGE MAPPING FOR THE PLACEHOLDER. **
+        translation: "/src/main/resources/assets/neoforge/lang/%locale_with_underscore%.%file_extension%"
+        dest: "/messages.%file_extension%"
```

Comment:
Uhh is that destination correct?

Reply:
Yes. :smile: I prefer not putting the whole directory tree in the Crowdin interface, as to reduce interface clutter with all the directories, to guard against possible relocations of the file in the future (however unlikely they may be), and to give them a neater name than just "`en_us.json`" in the interface.

(Note that, contrary to intuition, the `translation` field controls the actual destination of the _translated_ file(s) within the on-disk project/export archive; the `dest` field controls the destination of the _source_ file(s) in the Crowdin project/branch.)

However, that's my personal preference -- I could change it to following the directory structure, if there's good reasons to do so.

Reply:
Ohhh, neat!

Reply:
Just for better illustration, the NeoForged project at https://crowdin.neoforged.net/neoforge/en-GB shows how much cleaner it can be to view the messages when the intervening directories aren't included. (The title of the source file is customizable, but doesn't affect the actual source filename, which is still `messages.json` in that case.)